functions/Set-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,332 @@
+        [string]$OnFailAction,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($OnFailAction -ne 'GoToStep') -and ($_ -ge 1))
+            {
+                Throw "Parameter OnFailStepId can only be used with OnFailAction 'GoToStep'."
+            }
+            else 
+            {
+                $true    
+            }
+        })]
+        [int]$OnFailStepId,
+        [Parameter(Mandatory = $false)]
+        [string]$DatabaseName,
+        [Parameter(Mandatory = $false)]
+        [string]$DatabaseUserName,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryAttempts,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryInterval,
+        [Parameter(Mandatory = $false)]
+        [string]$OutputFileName,
+        [Parameter(Mandatory = $false)]
+        [string]$ProxyName,
+        [switch]$Silent
+    )
+
+    PROCESS
+    {
+        # Try connecting to the instance
+        Write-Message -Message "Attempting to connect to Sql Server.." -Level 2 -Silent $Silent
+        try 
+        {
+            $Server = Connect-SqlServer -SqlServer $SqlServer -SqlCredential $SqlCredential
+        }
+        catch 
+        {
+            Stop-Function -Message "Could not connect to Sql Server instance" -Silent $Silent -Target $SqlServer 
+            return
+        }
+
+        # Check if the job exists
+        if(($Server.JobServer.Jobs).Name -notcontains $JobName)
+        {
+            Stop-Function -Message "Job '$($JobName)' doesn't exists on '$($SqlServer)'" -Silent $Silent -Target $SqlServer 
+            return
+        }
+        else 
+        {
+            # Check if the job step exists
+            if(($Server.JobServer.Jobs[$JobName].JobSteps).Name -notcontains $StepName)
+            {
+                Stop-Function -Message "Step '$($StepName)' doesn't exists on '$($SqlServer)'" -Silent $Silent -Target $SqlServer 
+                return
+            }
+            else 
+            {
+                # Get the job step
+                $JobStep = $Server.JobServer.Jobs[$JobName].JobSteps[$StepName]
+
+                #region job step options
+                # Setting the options for the job step
+                if($NewName.Length -ge 1)
+                {
+                    Write-Message -Message "Setting job step name to '$($NewName)'" -Level 5 -Silent $Silent
+                    $JobStep.Rename($NewName)
+                }
+
+                if($Subsystem.Length -ge 1)
+                {
+                    Write-Message -Message "Setting job step subsystem to '$($Subsystem)'" -Level 5 -Silent $Silent
+                    $JobStep.Subsystem = $Subsystem
+                }
+
+                if($Command.Length -ge 1)
+                {
+                    Write-Message -Message "Setting job step command to '$($Command)'" -Level 5 -Silent $Silent
+                    $JobStep.Command = $Command
+                }
+
+                if($CmdExecSuccessCode -ge 0)
```

Comment:
Will also be applied *when the parameter is not bound*. Beware of that comparison, valuetypes are initialized with their default value when parameters are not bound (which in case of integer is '0'). Instead use this:
`if ($PSBoundParameters.ContainsKey("CmdExecSuccessCode"))`
In order to check whether a parameter was bound